### ISSUE TITLE
Support pkg-config and/or external library

### ISSUE THREAD

**Issue by gen2brain** [hybridgroup/gocv#126]
Currently I don't see LDFLAGS defined anywhere in the code, I had to export CGO_LDFLAGS in order to `go get` this package. 

```
export CGO_LDFLAGS="-lopencv_core -lopencv_videoio -lopencv_imgproc -lopencv_imgcodecs -lopencv_objdetect -lopencv_video -lopencv_highgui -lopencv_dnn -lopencv_features2d"
```
Instead of just instructions how to build OpenCV in specific way why not allow something like this `//#cgo pkg-config: opencv` , to be used with `-tags pkg-config`.  Or/and add all required libraries to LDFLAGS so it can be used with `-tags extlib` or something like that.

My pkg-config for example returns correct flags and gocv can be built with those:
```
# pkg-config --cflags --libs opencv
-I/usr/include/opencv -lopencv_dnn -lopencv_ml -lopencv_objdetect -lopencv_shape -lopencv_stitching -lopencv_superres -lopencv_videostab -lopencv_calib3d -lopencv_features2d -lopencv_highgui -lopencv_videoio -lopencv_imgcodecs -lopencv_video -lopencv_photo -lopencv_imgproc -lopencv_flann -lopencv_core
```

**Comment by willdurand** [comment]
I think it is a good idea.

**Comment by deadprogram** [comment]
Hello @gen2brain the main reason why we're not using pkg-config is so alternative version of OpenCV such as the Intel CV SDK can be easily built against. Hence the ENV vars being set via the `env.sh` and `env.cmd` scripts.

I'd be really happy to find a way to use pkg-config, but in all my experiments it prevents an alternative build of OpenCV being used.

**Comment by gen2brain** [comment]
I didn't mean to change defaults, just to add one more file, for example `gocv_pkgconfig.go` with this content:
```
// +build pkgconfig

//#cgo pkg-config: opencv
import "C"
```
And we can then use it like this `go get -tags pkgconfig gocv.io/x/gocv` in Linux, Windows, probably macOS with brew etc.  for when no special extensions are needed, just system default.

**Comment by deadprogram** [comment]
That seems like a good approach to me, @gen2brain

**Comment by deadprogram** [comment]
Was just thinking about this, and how it would be a better default than the current setup. We could have:

```go
// -build customenv

//#cgo pkg-config: opencv
import "C"
```

Now as long as you have used the "normal" way to install OpenCV alongside pkg-config you can just run using:

```
go run ./cmd/version/main.go
```

and then if we really do need the custom config, we can run or build using:

```
source ./env.sh
go run -tags=customenv ./cmd/version/main.go
```

This would help solve a lot of confusion for newcomers to the project, I think.

**Comment by deadprogram** [comment]
Of course, so far i have not been able to actually get that syntax to compile under Linux. Still investigating...

**Comment by deadprogram** [comment]
@gen2brain did you ever have a chance to do any more work on this idea?

**Comment by gen2brain** [comment]
@deadprogram I will try something these days, didn't have time to try.

**Comment by deadprogram** [comment]
This has now been implemented in the `dev` branch thanks to #210 I will close this issue once it has been released.

**Comment by deadprogram** [comment]
We just released v0.13.0 which includes this code. It should help remove a lot of confusion when getting started with GoCV.

I am going to close this issue now, please reopen if needed. Thank you!
